### DISCUSSION TITLE
Question on the Linux installer name

### DISCUSSION THREAD

**Discussion by freixas** [wiverson/maven-jpackage-template#64] [Q&A]
Right now, my Debian package name is gamma_1.0.0-1_amd64.deb. Everything else is gamma-1.0.0. I'm not too worried about the dash vs. underscore difference, but I dislike the -1_amd64 part.

This could be in my code, in your jtoolprovider plugin, or, most likely, one of the many undocumented quirks in jpackager. My packaging file for Linux isn't much different than the ones for Windows or Mac (see https://github.com/freixas/gamma).

Looking at your assets, I see you have the same problem, so my money is on a jpackage quirk, but I thought I'd ask.

BTW, I now have a MacOS build (thanks to VirtualBox). Thanks for your tip on getting the file out of quarantine when it's unsigned!

**Comment by wiverson** [comment]
AFAIK you can change the name of the built installer to whatever you want with a simple file rename. I don't think it affects the signing or stapling.

**Reply by freixas** [reply]
Since this is the Linux Debian installer, it's definitely not affected by signing or stapling. I just peeked inside the Debian file and couldn't find the "-1_amd64" anywhere. 

I assume you're in the same boat as me and you don't know where the "-1_amd64" is coming from. Except since you authored the jtoolprovider plugin, you're probably able to exclude that from the list of options. I've checked out my code--if it were in my code, the windows and mac installers would also have the suffix. So I think we can say that this is coming from jpackage.

Narrowing down the possible culprits helped. I did some searching and found a guide on [Debian package naming](https://www.debian.org/doc/manuals/debian-reference/ch02.en.html#_debian_package_file_names). So ""-1" is the Debian revision. Hunting around a bit, I learned that it is optional (but the architecture is not) and indicates changes to certain parts of the Debian package, as opposed to changes in the thing being packaged. The concept might apply, for example, if you made some changes to the jpackage options, but not to the Java code. The software version wouldn't change, but the Debian revision should increment. I suspect jpackage always generates "-1", though.

Mystery solved.
